cmd/otel-agent/subcommands/run/command.go

```diff
@@ -174,19 +179,25 @@ func runOTelAgentCommand(ctx context.Context, params *subcommands.GlobalParams,
 		// ctx is required to be supplied from here, as Windows needs to inject its own context
 		// to allow the agent to work as a service.
 		fx.Provide(func() context.Context { return ctx }), // fx.Supply(ctx) fails with a missing type error.
+
+		fx.Provide(func(deps traceconfig.Dependencies, _ optional.Option[configsync.Component]) (traceconfig.Component, error) {
```

Comment:
`_ optional.Option[configsync.Component]` If we are not using we can remove from the function parameters

Reply:
I guess `traceconfig.NewConfig` must be created after `configsync.Component` so this parameter is probably needed. In order to better express this dependency, I think  `traceconfig.NewConfig` should take `configsync.Component` and document this dependency.

Question: `configsync.Component` is optional, can `traceconfig.NewConfig` proper handles `None` value?

Reply:
No we can't, this is for dependency ordering and is required.

Reply:
As you can imagine, `traceconfig` itself doesn't use `configsync`; that component interacts directly with the core config. There's a bunch of follow-up work for config propagation and _actually_ allowing propagating config changes to trace-config and other places in the `otel-agent` but that's something for later.

Would you be OK with a `TODO` comment and a card in our backlog for this?

Reply:
Yes sure

Reply:
Thanks for the clarification @truthbk @ogaca-dd